### ISSUE TITLE
In the particle shader the deterministic sequence of random values is not guaranteed

### ISSUE THREAD

**Issue by greycheeked** [godotengine/godot#80625]
### Godot version

v4.1.1.stable.official [bd6af8e0e]

### System information

Windows 10.0.22621 - Vulkan (Forward+) - dedicated NVIDIA GeForce GTX 1650 SUPER (NVIDIA; 31.0.15.3130) - Intel(R) Core(TM) i5-10400F CPU @ 2.90GHz (12 Threads)

### Issue description

I think I found a bug in the particle shader. It revolves around the following line:

```
// apply linear acceleration
force += length(VELOCITY) > 0.0 ? normalize(VELOCITY) * tex_linear_accel * mix(linear_accel_min, linear_accel_max, rand_from_seed(alt_seed)) : vec3(0.0);

```
As long as VELOCITY is greater than 0, "rand_from_seed(alt_seed)" is called among other things. Thereby "alt_seed" is changed.
But if now damping is active, VELOCITY reaches 0 at some point. Then "rand_from_seed(alt_seed)" is NOT called anymore.
That means: all following random values, which depend on "rand_from_seed()", change once abruptly.

### Steps to reproduce

The error probably manifests itself in very different ways. As soon as damping brings the speed to 0, there will be one-time jumpy changes in all aspects that use edge_fom_seed:

- radial acceleration
- tangential acceleration
- orbit velocity
- damping
- angular velocity
- animation offset and speed
- hue

### Minimal reproduction project

N/A

**Comment by clayjohn** [comment]
Good catch! This random values should be calculated outside the conditional so that it is always calculated and other random values don't change based on the presence/absence of the condition.
